### DISCUSSION TITLE
Cannot fetch changelog if URL is HTTPS

### DISCUSSION THREAD

**Discussion by nhmarujo** [renovatebot/renovate#23207] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.159.6

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Since https://gitlab.com/renovate-bot/renovate-runner/-/merge_requests/1873 my projects are not able to fetch changelogs. I get a message on the logs like `Invalid gitlab URL found: https://gitlab.com/xxx"`

I believe this is due to https://github.com/renovatebot/renovate/pull/23128, namely because the function being now used is assuming by default that only http is allowed, as can be seen [here](https://github.com/renovatebot/renovate/blob/main/lib/util/url.ts#L86).

Could it be that it needs to be called as `validateUrl(word, false)` in [here](https://github.com/setchy/renovate/blob/a1e90ece14de5284feddcaecff78f1756c15daa6/lib/workers/repository/update/pr/changelog/release-notes.ts#L336) and [here](https://github.com/setchy/renovate/blob/a1e90ece14de5284feddcaecff78f1756c15daa6/lib/workers/repository/update/pr/changelog/release-notes.ts#L341)?

### Relevant debug logs

<details><summary>Logs</summary>

```
{"name":"renovate","hostname":"runner-XXX","pid":34,"level":20,"logContext":"rlGfe1p5e_YY5pPlK5_BX","repository":"XXX","branch":"XXX","msg":"Fetching changelog: https://gitlab.com/XXX (3.1.1-2249 -> 3.1.1-2259)","time":"2023-07-05T21:46:12.763Z","v":0}
{"name":"renovate","hostname":"runner-XXX","pid":34,"level":20,"logContext":"rlGfe1p5e_YY5pPlK5_BX","repository":"XXX","branch":"XXX","msg":"Invalid gitlab URL found: https://gitlab.com/XXX","time":"2023-07-05T21:46:12.763Z","v":0}
```

</details>
Note: Add to obfuscate some info `XXX`

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by viceice** [comment, marked as answer]
I see now, you need to update again, see https://gitlab.com/renovate-bot/renovate-runner/-/merge_requests/1874 and https://gitlab.com/renovate-bot/renovate-runner/-/releases/v14.64.4

**Comment by setchy** [comment]
See https://github.com/renovatebot/renovate/discussions/23156#discussioncomment-6370203, resolved in [v35.159.7](https://github.com/renovatebot/renovate/releases/tag/35.159.7)

**Reply by nhmarujo** [reply]
I forced it to run again and this time it picked version [v35.159.7](https://github.com/renovatebot/renovate/releases/tag/35.159.7) and I can confirmed that it worked now. Thank you!
